src/utils/request.ts

```diff
@@ -9,19 +13,50 @@ export const post = async (url: string, body: unknown) => {
       credentials: "include",
     });
     if (!response.ok) {
-      throw new Error(await getErrorMessage(response, "POST"));
+      throw new Error(getErrorMessage(response, "POST"));
     }
     return response;
   } catch (e: any) {
     handleError(e);
   }
 };
 
-const getErrorMessage = async (response: Response, method: string) => {
+const getErrorMessage = (response: Response, method: string) => {
   const { status, statusText } = response;
   return `${method} Error: ${status} - ${statusText}`;
 };
 
 const handleError = (error: string) => {
   reportError(new Error(error)).warning();
 };
+
+export const fetchWithRetry = <T = any>(
+  url: string,
+  options: RequestInit,
+  retries: number = 3,
+  backoff: number = 150,
+): Promise<T> =>
```

Comment:
I think the typing for this should actually be 
```suggestion
): Promise<{ data: T }> =>
```